primitives/src/merkle_tree/internal.rs

```diff
@@ -137,29 +135,24 @@ impl<T: NodeValue> MerkleCommitment<T> for MerkleTreeCommitment<T> {
 #[derivative(Eq, Hash, PartialEq)]
 #[serde(bound = "E: CanonicalSerialize + CanonicalDeserialize,
              I: CanonicalSerialize + CanonicalDeserialize,")]
-pub struct MerkleProof<E, I, T, Arity>
+pub struct MerkleProof<E, I, T, const ARITY: usize>
```

Comment:
i believe the [naming convention](https://rust-lang.github.io/api-guidelines/naming.html#casing-conforms-to-rfc-430-c-case) for types are UpperCamelCase, even with constant generic

Reply:
lint or fmt was complaining. Looks like most people are using uppercase, e.g. https://www.reddit.com/r/rust/comments/11medef/rust_const_generics_for_proof_types/

Reply:
I see, you will get the following linter warning instead: 
```
  |
1 | trait Foo<E, const Arity: usize> 
  |                          ^^^^^ help: convert the identifier to upper case: `ARITY`
  |
  = note: `#[warn(non_upper_case_globals)]` on by default
```